public_html/admin/controller/responses/setting/setting_quick_form.php

```diff
@@ -240,40 +241,30 @@ public function quick_start_save_next()
                 $this->loadLanguage('common/quick_start');
+                $post = $this->request->post;
+                $section = $group;
 
                 if ($group == 'appearance') {
                     $section = $this->request->get['tmpl_id'] == 'default'
-                        ? 'appearance'
-                        : $this->request->get['tmpl_id'];
-                    if (has_value($this->request->post['config_logo'])) {
-                        $this->request->post['config_logo'] = html_entity_decode(
-                            $this->request->post['config_logo'],
-                            ENT_COMPAT,
-                            'UTF-8'
-                        );
-                    } elseif (!$this->request->post['config_logo']
-                        && isset($this->request->post['config_logo_resource_id'])
-                    ) {
-                        //we save resource ID vs resource path
-                        $this->request->post['config_logo'] = $this->request->post['config_logo_resource_id'];
+                                ? 'appearance'
+                                : $this->request->get['tmpl_id'];
+
+                    foreach (['config_logo', 'config_mail_logo', 'config_icon'] as $n) {
+                        $newName = in_array($n, ['config_logo', 'config_mail_logo']) ? $n.'_'.$languageId : $n;
+                        if (isset($post[$n.'_resource_id'])) {
+                            //we save resource ID vs resource path
+                            $post[$newName] = $post[$n.'_resource_id'];
+                        }elseif (isset($post[$n])) {
+                            $post[$newName] = html_entity_decode($post[$n], ENT_COMPAT, 'UTF-8');
+                            if($newName != $n){
+                                unset($post[$n]);
+                            }
+                        }
                     }
-
-                    if (has_value($this->request->post['config_icon'])) {
-                        $this->request->post['config_icon'] = html_entity_decode(
-                            $this->request->post['config_icon'],
-                            ENT_COMPAT,
-                            'UTF-8'
-                        );
-                    } elseif (!$this->request->post['config_icon']
-                        && isset($this->request->post['config_icon_resource_id'])
-                    ) {
-                        //we save resource ID vs resource path
-                        $this->request->post['config_icon'] = $this->request->post['config_icon_resource_id'];
-                    }
-                    $this->model_setting_setting->editSetting($section, $this->request->post, $store_id);
-                } else {
-                    $this->model_setting_setting->editSetting($group, $this->request->post, $store_id);
                 }
+
+                $this->model_setting_setting->editSetting($section, $post, $store_id);
```

Comment:
Was this update confirmed by user? Need a code comment here. Why do we update and what we update?